### PR TITLE
Callable

### PR DESCRIPTION
**Pull request by Daniluko**

### CODE REVIEW

File: HomeWork/src/main/java/Callable/Main.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Created by Rostyslav on 17.01.19
+ */
+public class Main {
+    public static void main(String[] args) {
+        Resource res = new Resource(0);
+        ExecutorService executorService = Executors.newFixedThreadPool(3);
+
+        List<Future<String>> list = new ArrayList<Future<String>>();
+        Callable<String> callable = new ThreadTest("test1",res);
```
Comment (un-install) [inline HomeWork/src/main/java/Callable/Main.java]: Why you use String in generics?

File: HomeWork/src/main/java/Callable/Resource.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package Callable;
+
+/**
+ * Created by Rostyslav on 17.01.19
+ */
+public class Resource  {
+    private int value = 0;
```
Comment (un-install) [inline HomeWork/src/main/java/Callable/Resource.java]: You need to use atomic to make it threadsafe, or make updating methods synchronized

File: HomeWork/src/main/java/Callable/ThreadTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        this.res = res;
+    }
+
+
+    public String call() throws Exception {
+        try {
+            System.out.println(name + " waiting");
+            for (int i = 0 ; i < 5 ; i++){
+                res.increment();
+                Thread.sleep(100);
```
Comment (un-install) [inline HomeWork/src/main/java/Callable/ThreadTest.java]: Also. Why you return String, if your priority data is Resource. Is it for masking bugs?)

File: HomeWork/src/main/java/Callable/Main.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+        for (int i =0 ; i < 3; i++){
+            Future<String> future = executorService.submit(callable);
```
Comment (un-install) [inline HomeWork/src/main/java/Callable/Main.java]: You dont need to create future object of future, just call method when you add it to list

File: HomeWork/src/main/java/Callable/ThreadTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    private String name;
+    private Resource res;
+
+    public ThreadTest(String name, Resource res) {
+        this.name = name;
+        this.res = res;
+    }
+
+
+    public String call() throws Exception {
```
Comment (SergeyPasko) [inline HomeWork/src/main/java/Callable/ThreadTest.java]: Bad. Use concretical exception

**Pull request merged**